File: moreapps/src/main/java/io/github/raghavsatyadev/moreapps/settings/PeriodicUpdateSettings.java
package io.github.raghavsatyadev.moreapps.settings;

import androidx.annotation.DrawableRes;

import java.util.concurrent.TimeUnit;

public class PeriodicUpdateSettings {
    private int interval;
    private TimeUnit timeUnit;
    @DrawableRes
    private int bigIconID;
    @DrawableRes
    private int smallIconID;

    public PeriodicUpdateSettings() {
        interval = 7;
        timeUnit = TimeUnit.DAYS;
    }

    public int getInterval() {
        return interval;
    }

    public TimeUnit getTimeUnit() {
        return timeUnit;
    }

    public int getBigIconID() {
        return bigIconID;
    }

    public int getSmallIconID() {
        return smallIconID;
    }

    public void setSettings(int interval, TimeUnit timeUnit, int bigIconID, int smallIconID) {
        this.interval = interval;
        this.timeUnit = timeUnit;
        this.bigIconID = bigIconID;
        this.smallIconID = smallIconID;
    }
}

File: moreapps/src/main/java/io/github/raghavsatyadev/moreapps/settings/PeriodicUpdateSettings.kt
package io.github.raghavsatyadev.moreapps.settings

import androidx.annotation.DrawableRes
import java.util.concurrent.TimeUnit
import java.util.concurrent.TimeUnit.DAYS

class PeriodicUpdateSettings {
    var interval = 7
        private set
    var timeUnit: TimeUnit
        private set

    @DrawableRes
    var bigIconID = 0
        private set

    @DrawableRes
    var smallIconID = 0
        private set

    init {
        timeUnit = DAYS
    }

    fun setSettings(interval: Int, timeUnit: TimeUnit, bigIconID: Int, smallIconID: Int) {
        this.interval = interval
        this.timeUnit = timeUnit
        this.bigIconID = bigIconID
        this.smallIconID = smallIconID
    }
}